cmd/clusterctl/client/cluster/mover.go

```diff
@@ -1164,3 +1173,35 @@ func (o *objectMover) checkTargetProviders(toInventory InventoryClient) error {
 
 	return kerrors.NewAggregate(errList)
 }
+
+func patchTopologyManagedFields(ctx context.Context, oldManagedFields []metav1.ManagedFieldsEntry, obj *unstructured.Unstructured, cTo client.Client) error {
+	var containsTopologyManagedFields bool
+	// managedFields will contain the target managed fields obj gets patched to.
+	wantManagedFields := []metav1.ManagedFieldsEntry{}
+
+	for _, m := range oldManagedFields {
+		if m.Manager == "manager" && m.Operation == metav1.ManagedFieldsOperationUpdate && m.Subresource == "" {
+			continue
+		}
+		if m.Operation == metav1.ManagedFieldsOperationApply &&
+			m.Manager == structuredmerge.TopologyManagerName &&
+			m.Subresource == "" {
+			containsTopologyManagedFields = true
+		}
+		wantManagedFields = append(wantManagedFields, m)
+	}
+
+	// Return early if the managed fields had no entry from topology manager
+	if !containsTopologyManagedFields {
+		return nil
+	}
+
+	base := obj.DeepCopy()
+	obj.SetManagedFields(wantManagedFields)
```

Comment:
```suggestion
// patchTopologyManagedFields patches the managed fields of obj if parts of it are owned by the topology controller.
// This is necessary to ensure the managed fields created by the topology controller are still present and thus to
// prevent unnecessary machine rollouts.
func patchTopologyManagedFields(ctx context.Context, oldManagedFields []metav1.ManagedFieldsEntry, obj *unstructured.Unstructured, cTo client.Client) error {
	var containsTopologyManagedFields bool
	// managedFields will contain the target managed fields obj gets patched to.
	wantManagedFields := []metav1.ManagedFieldsEntry{}

	for _, m := range oldManagedFields {
		// Check if the object was owned by the topology controller.
		if m.Operation == metav1.ManagedFieldsOperationApply &&
			m.Manager == structuredmerge.TopologyManagerName &&
			m.Subresource == "" {
			containsTopologyManagedFields = true
		}
		// Filter out the entry which only contains ownership of an ownerReference which does not exist anymore because the
		// uid of the owner changed during move.
		if m.Manager == "manager" && m.Operation == metav1.ManagedFieldsOperationUpdate && m.Subresource == "" {
			continue
		}
		wantManagedFields = append(wantManagedFields, m)
	}

	// Return early if the object was not owned by the topology controller.
	if !containsTopologyManagedFields {
		return nil
	}

	base := obj.DeepCopy()
	obj.SetManagedFields(wantManagedFields)
```

Reply:
Alternative without filtering the "misleading" managed field for ownerreference, but to stay consistent:

```suggestion
// patchTopologyManagedFields patches the managed fields of obj if parts of it are owned by the topology controller.
// This is necessary to ensure the managed fields created by the topology controller are still present and thus to
// prevent unnecessary machine rollouts.
func patchTopologyManagedFields(ctx context.Context, oldManagedFields []metav1.ManagedFieldsEntry, obj *unstructured.Unstructured, cTo client.Client) error {
	var containsTopologyManagedFields bool
	// Check if the object was owned by the topology controller.
	for _, m := range oldManagedFields {
		if m.Operation == metav1.ManagedFieldsOperationApply &&
			m.Manager == structuredmerge.TopologyManagerName &&
			m.Subresource == "" {
			containsTopologyManagedFields = true
			break
		}
	}

	// Return early if the object was not owned by the topology controller.
	if !containsTopologyManagedFields {
		return nil
	}

	base := obj.DeepCopy()
	obj.SetManagedFields(oldManagedFields)
```